### ISSUE TITLE
Chief process does not gracefully exits 

### ISSUE THREAD

**Issue by sergiigladchuk** [keras-team/keras-tuner#254]
Hi,
I am really new to the keras-tuner, but find it very useful. I was playing with "Distributed Tuning" tutorial from here (https://keras-team.github.io/keras-tuner/tutorials/distributed-tuning/) and chief process never exits when all the tuning options are tested. Is it by design?
Or am I missing something here? Should chief process exit when all the tuning options are exhausted?

I have checked `oracle_chief.py` there is just loop at the end:
```
server.start()
    while True:
        # The server does not block otherwise.
        time.sleep(10)
```
No exception catch to stop gRPC server and make it possible to gracefully exit. 

I ran same example from tutorial with reduced number of options for hyperparameters to try.

**Comment by scfox** [comment]
I am seeing this same behavior.  After all tuners complete, they exit but chief process continues to run.

**Comment by sergiigladchuk** [comment]
> I am seeing this same behavior. After all tuners complete, they exit but chief process continues to run.

Hopefully, maintainers will find time to help me to fix my PR and move the it forward. It is supposed to add at least possibility to send INT and other signals.

**Comment by haifeng-jin** [comment]
Thank you for the issue.
This problem is fixed in the latest release of 1.0.2.
Let me know if it doesn't.
